manifests/lvs/real_server.pp

```diff
@@ -31,19 +31,13 @@
 #     }
 #
 define keepalived::lvs::real_server (
-  $virtual_server,
-  $ip_address,
-  $port,
-  $options = {},
+  String[1] $virtual_server,
+  Stdlib::IP::Address $ip_address,
+  Stdlib::Port $port,
+  Hash $options = {},
```

Comment:
Looking at https://github.com/voxpupuli/puppet-keepalived/blob/master/templates/_format_options.erb it looks like we can have a defined type `type Keepalived::Options = Hash[String, Variant[Hash, String, Boolean]]`. Technically the latter Hash is a `Keepalived::Options` but I don't know if Puppet supports recursive types.

Reply:
I think it'd need to be
`Keepalived::Options = Hash[String, Variant[Hash, String, Boolean, Integer]]`
https://github.com/voxpupuli/puppet-keepalived/blob/f572a800f803e940826de62a071684bfe04d86fd/spec/defines/keepalived_lvs_real_server_spec.rb#L67

Not sure it's worth it.

Reply:
`Hash[String, Any]` then?

Reply:
Sure, ok.  I'd imagine 90% of the times where we're currently using `Hash` we could use this instead.  It's probably quite rare to use anything other than a string as the hash key.

The primary aim of this PR was just to get rid of the validate functions.  There are a few classes that still have no types at all.